server/api/routers/inference/services.py

```diff
@@ -0,0 +1,190 @@
+from typing import Dict, Optional, List, Any, Tuple
+from atomic_agents import BasicChatInputSchema
+from .models import ToolResult, IntegrationType, ProjectAction
+from .analyzers import MessageAnalyzer, ResponseAnalyzer
+from .factories import AgentFactory, ModelManager
+from core.config import settings
+
+# Store agent instances to maintain conversation context
+agent_sessions: Dict[str, Any] = {}
+
+class ToolExecutor:
+    """Handles tool execution (both native and manual)"""
+    
+    @staticmethod
+    def execute_manual(message: str) -> ToolResult:
+        """Manually execute tool based on message analysis"""
+        try:
+            from tools.projects_tool.tool import ProjectsTool, ProjectsToolInput
+            projects_tool = ProjectsTool()
+            action = MessageAnalyzer.determine_action(message)
```

Comment:
Do you think this will be brittle, especially over time? Seems like it's possible the incoming message could be phrased in various ways that we might not catch. We could probably make another inference call here and ask the model to pick an action, namespace, etc from a list of options to make it more flexible.

I think we could also grab namespaces and project IDs from incoming request fields and not include them in the message at all.